src/wrappers/nats.js

```diff
@@ -0,0 +1,205 @@
+            tracer.sendTrace(() => {});
+        }
+        tracer.addRunner(nodeEvent, runnerResult);
+    } catch (err) {
+        tracer.addException(err);
+    }
+    // Throwing error in case of sync user function.
+    if (originalHandlerSyncErr) {
+        throw originalHandlerSyncErr;
+    }
+    return runnerResult;
+}
+
+/**
+ * Wraps nats subscribe function with tracing.
+ * @param {Function} wrappedFunction nats subscribe function.
+ * @param {string} serverHostname nats server host name.
+ * @return {Function} subscribe function rsoponse.
+ */
+function natsSubscribeWrapper(wrappedFunction, serverHostname) {
+    return function internalNatsSubscribeWrapper(subject, opts, callback) {
+        let clientRequest;
+        const { opts_internal, callback_internal } = getSubscribeParams(opts, callback);
+        let patchedCallback = callback_internal;
+        try {
+            const isRequestCall = isNatsRequestCall(subject);
+            patchedCallback = (callback_msg, callback_reply, callback_subject, callback_sid) => {
+                traceContext.RunInContext(
+                    tracer.createTracer,
+                    () => natsSubscribeCallbackMiddleware(
+                        callback_msg,
+                        callback_reply,
+                        callback_subject,
+                        callback_sid,
+                        callback_internal,
+                        serverHostname,
+                        isRequestCall
+                    )
+                );
+            };
+            clientRequest = wrappedFunction.apply(this, [subject, opts_internal, patchedCallback]);
+        } catch (err) {
+            if (!clientRequest) {
+                clientRequest = wrappedFunction.apply(this, [subject, opts, callback]);
+            }
+        }
+        return clientRequest;
+    };
+}
+
+/**
+ * Wraps nats connect function.
+ * @param {Function} connectFunction nats connect function.
+ * @return {Function} nats connect function response.
+ */
+function natsConnectWrapper(connectFunction) {
+    traceContext.init();
+    tracer.getTrace = traceContext.get;
+    return function internalNatsConnectWrapper(url, opts) {
+        const connectFunctionResponse = connectFunction(url, opts);
+        if (connectFunctionResponse && connectFunctionResponse.constructor) {
+            if (connectFunctionResponse.constructor.name !== NATS_TYPES.mainWrappedFunction) {
+                return connectFunctionResponse;
+            }
+            const serverHostname = getServerHostname(connectFunctionResponse.currentServer);
+            shimmer.wrap(connectFunctionResponse, 'subscribe', () => natsSubscribeWrapper(connectFunctionResponse.subscribe, serverHostname));
+        }
+        return connectFunctionResponse;
+    };
+}
+
+module.exports = {
+    /**
+     * Initializes the nats tracer
+     */
+    init() {
+        const nats = tryRequire('nats');
```

Comment:
Why are you tryRequiring? Use the module utils?

Reply:
we don't export module utils.